File: app/src/main/java/com/watnapp/etipitaka/plus/model/HistoryItem.java
package com.watnapp.etipitaka.plus.model;
import android.provider.BaseColumns;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

public class HistoryItem extends ModelBase {
  private Context context;
  private int id;
  private int historyId;
  private int volume;
  private int page;
  private int status;

  public HistoryItem() {
    super();
  }

  @Override
  public int getId() {
    return id;
  }

  @Override
  public void setId(int id) {
    this.id = id;
  }

  public int getHistoryId() {
    return historyId;
  }

  public void setHistoryId(int historyId) {
    this.historyId = historyId;
  }

  public int getVolume() {
    return volume;
  }

  public void setVolume(int volume) {
    this.volume = volume;
  }

  public int getPage() {
    return page;
  }

  public void setPage(int page) {
    this.page = page;
  }

  public Status getStatus() {
    return Status.values()[status];
  }

  public void setStatus(Status status) {
    this.status = status.ordinal();
  }

  @Override
  public void fromCursor(Cursor cursor, Context context) {
    this.id = cursor.getInt(cursor.getColumnIndex(BaseColumns._ID));
    this.historyId = cursor.getInt(cursor.getColumnIndex(HistoryItemTable.HistoryItemColumns.HISTORY_ID));
    this.volume = cursor.getInt(cursor.getColumnIndex(HistoryItemTable.HistoryItemColumns.VOLUME));
    this.page = cursor.getInt(cursor.getColumnIndex(HistoryItemTable.HistoryItemColumns.PAGE));
    this.status = cursor.getInt(cursor.getColumnIndex(HistoryItemTable.HistoryItemColumns.STATUS));
    this.context = context;
  }

  @Override
  public ContentValues toContentValues() {
    ContentValues values = new ContentValues();
    values.put(HistoryItemTable.HistoryItemColumns.HISTORY_ID, this.historyId);
    values.put(HistoryItemTable.HistoryItemColumns.VOLUME, this.volume);
    values.put(HistoryItemTable.HistoryItemColumns.PAGE, this.page);
    values.put(HistoryItemTable.HistoryItemColumns.STATUS, this.status);
    return values;
  }

  public static HistoryItem newInstance(Cursor cursor, Context context) {
    HistoryItem historyItem = new HistoryItem();
    historyItem.fromCursor(cursor, context);
    return historyItem;
  }

  public enum Status {
    NONE(0), READ(1), SKIMMED(2);

    private int code;

    private Status(int code) {
      this.code = code;
    }

    public int getCode() {
      return code;
    }
  }
}

File: app/src/main/java/com/watnapp/etipitaka/plus/model/HistoryItem.kt
package com.watnapp.etipitaka.plus.model

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.provider.BaseColumns

class HistoryItem : ModelBase() {
    private var id = 0
    private var historyId = 0
    private var volume = 0
    private var page = 0
    private var status = 0

    override fun getId(): Int = id

    override fun setId(id: Int) {
        this.id = id
    }

    fun getHistoryId(): Int = historyId

    fun setHistoryId(historyId: Int) {
        this.historyId = historyId
    }

    fun getVolume(): Int = volume

    fun setVolume(volume: Int) {
        this.volume = volume
    }

    fun getPage(): Int = page

    fun setPage(page: Int) {
        this.page = page
    }

    fun getStatus(): Status = Status.values()[status]

    fun setStatus(status: Status) {
        this.status = status.ordinal
    }

    override fun fromCursor(cursor: Cursor, context: Context) {
        id = cursor.getInt(cursor.getColumnIndexOrThrow(BaseColumns._ID))
        historyId = cursor.getInt(cursor.getColumnIndexOrThrow(HistoryItemTable.HistoryItemColumns.HISTORY_ID))
        volume = cursor.getInt(cursor.getColumnIndexOrThrow(HistoryItemTable.HistoryItemColumns.VOLUME))
        page = cursor.getInt(cursor.getColumnIndexOrThrow(HistoryItemTable.HistoryItemColumns.PAGE))
        status = cursor.getInt(cursor.getColumnIndexOrThrow(HistoryItemTable.HistoryItemColumns.STATUS))
    }

    override fun toContentValues(): ContentValues =
        ContentValues().apply {
            put(HistoryItemTable.HistoryItemColumns.HISTORY_ID, historyId)
            put(HistoryItemTable.HistoryItemColumns.VOLUME, volume)
            put(HistoryItemTable.HistoryItemColumns.PAGE, page)
            put(HistoryItemTable.HistoryItemColumns.STATUS, status)
        }

    enum class Status(val code: Int) {
        NONE(0),
        READ(1),
        SKIMMED(2)
    }

    companion object {
        @JvmStatic
        fun newInstance(cursor: Cursor, context: Context): HistoryItem =
            HistoryItem().apply { fromCursor(cursor, context) }
    }
}
